tasks/test_e2e.yml

```diff
@@ -0,0 +1,60 @@
+- name: Querying metrics in Prometheus
+  hosts: undercloud-0
+  become: true
+  become_method: sudo
+  become_user: root
+  tasks:
+    - name: "Extract URL from stf-connectors template"
+      shell: grep interconnect /home/stack/virt/stf-connectors-osp16-edge.yaml | sed 's;.*\(-service-telemetry.*\);prometheus-operated\1;'
+      register: prom_url
+      failed_when: "prom_url.stdout_lines|length == 0"
+    - debug:
+        var: prom_url.stdout_lines
+    
+    - debug: 
+        msg: The URL is {{ prom_url.stdout_lines }}
+
+    - name: "Capture ceph_cpu_bytes metrics"
+      shell: /usr/bin/curl -k http://{{ prom_url.stdout_lines[0] }}/api/v1/query? --data-urlencode 'query=collectd_ceph_ceph_bytes {plugin_instance="ceph-osd.0"}[1m]' > /tmp/query_ceph_cpu_bytes
+      register: checkmyconf
+      failed_when:
+        - checkmyconf.rc !=0 
+    
+    - name: "Capture collectd_cpu_percent"
+      shell: /usr/bin/curl -k http://{{ prom_url.stdout_lines[0] }}/api/v1/query? --data-urlencode 'query=collectd_cpu_percent {plugin_instance="0",type_instance="user",service="default-cloud1-coll-meter-smartgateway"}[1m]' > /tmp/query_collectd_cpu_percent
+      register: checkmyconf
+      failed_when:
+        - checkmyconf.rc !=0
+    
+    - name: "Capture collectd_iinterface_tx_total"
+      shell: /usr/bin/curl -k http://{{ prom_url.stdout_lines[0] }}/api/v1/query? --data-urlencode 'query=collectd_interface_if_packets_tx_total {type_instance="base"}[1m]' > /tmp/query_collectd_interface_tx_total
+      register: checkmyconf
+      failed_when:
+        - checkmyconf.rc !=0
+
+    - name: "[Test] Check whether ceph cpu  metrics received in prometheus for all ceph nodes"
```

Comment:
This says "**all** ceph nodes" but the grep looks like it would be satisfied by metrics from "**any** ceph node". Checking from All would be preferable - is that really what's happening here? I haven't looked at the output in a live system - could we count the number of lines in that file and expect it to match the number of ceph nodes?

Reply:
> This says "**all** ceph nodes" but the grep looks like it would be satisfied by metrics from "**any** ceph node". Checking from All would be preferable - is that really what's happening here? I haven't looked at the output in a live system - could we count the number of lines in that file and expect it to match the number of ceph nodes?

Count  lines in the query output won't help since it's not related to a number of ceph nodes.
Yea,you are right grep will satisfy "any". The queries are based on the known number of overcloud nodes.
For example  I know in advance how many overcloud nodes will be sending metrics e.g. 3 controllers,2 computes and 1 ceph.  Probably there is a way to check how many overcloud nodes we have before running the query and dynamically update query according to every time we run it it but I think for now it's enough. Let's have something and than will think how to enhance it.